Setup MIN_TIMESTAMP updater

elusive-m opened the pull request:
Uses the [create-pull-request GitHub action by Peter Evans](https://github.com/peter-evans/create-pull-request) to create pull requests. As per the discussion in #51, the workflow runs every month but this can be changed.

Closes #51

lberrymage commented:
bors r+

elusive-m committed:
Setup MIN_TIMESTAMP updater

Uses GitHub Actions to update the `MIN_TIMESTAMP` every month, in the
form of pull requests.

lberrymage commented on .github/workflows/update-timestamp.yaml:
```diff
@@ -0,0 +1,28 @@
+        env:
+          FILE: app/src/main/java/app/accrescent/client/data/Constants.kt
+          CONSTANT: 'const val MIN_TIMESTAMP: Long'
+        run: sed -i -E "s/^$CONSTANT = [[:digit:]]*$/$CONSTANT = $(date +'%s')/" $FILE
```
This will update `MIN_TIMESTAMP` to the Unix timestamp of the job runner. If the `timestamp` field of the [signed repository metadata](https://store.accrescent.app/repodata.0.json) is less than that, the client will be unable to fetch new metadata unless `timestamp` is reduced (which shouldn't be done if at all possible to preserve the value of downgrade protection).

Could you modify this to update the timestamp to the latest published value instead? It can be fetched via
`curl -s https://store.accrescent.app/repodata.0.json | jq '.timestamp'`.

elusive-m replied:
Done.

lberrymage commented on .github/workflows/update-timestamp.yaml:
```diff
@@ -0,0 +1,28 @@
+          CONSTANT: 'const val MIN_TIMESTAMP: Long'
+        run: sed -i -E "s/^$CONSTANT = [[:digit:]]*$/$CONSTANT = $(date +'%s')/" $FILE
+      - name: Create pull request
+        uses: peter-evans/create-pull-request@v4.1.1
```
nit: Specifying only the major version is fine and more consistent with existing Actions.

elusive-m replied:
Done.

lberrymage commented on .github/workflows/update-timestamp.yaml:
```diff
@@ -0,0 +1,28 @@
+name: Update MIN_TIMESTAMP
```
nit: To be consistent with previous commits, could you change this to "Bump" instead of "Update?"

elusive-m replied:
Done.

lberrymage commented on .github/workflows/update-timestamp.yaml:
```diff
@@ -0,0 +1,28 @@
+      - name: Create pull request
+        uses: peter-evans/create-pull-request@v4.1.1
+        env:
+          message: Update MIN_TIMESTAMP
```
Same as above: "Bump" instead of "Update" would be preferred.

elusive-m replied:
Done. I've also renamed the file to `bump-timestamp.yaml`.

lberrymage reviewed (changes requested):
First of all, thank you! Just a few minor suggestions.

elusive-m committed:
Reword "Update" to "Bump" in timestamp workflow

- Renamed `update-timestamp.yaml` to `bump-timestamp.yaml`
- Replaced relevant occurences of the word "Update" with "Bump" in the
  aforementioned file.

These changes ensure consistency with the previous commit messages.

elusive-m committed:
Use the latest v4 create-pull-request action

...instead of pinning it to a specific v4.x.y version in the timestamp
workflow. This new behavior is in line with the existing GitHub Actions.

elusive-m committed:
Fetch the latest timestamp from accrescent.app

lberrymage reviewed (approved):
Looks good to me! Thanks for the PR.
